### ISSUE TITLE
Generic pubsub over Matrix - room participant privacy?

### ISSUE THREAD

**Issue by jfrederickson** [matrix-org/matrix-doc#951]
I was thinking about using Matrix as a generic pubsub mechanism, and I realized that the only required change would be the ability for rooms to not publish their participants in the room state. After doing so, users could "subscribe" to such a room by simply joining it.

So for example, users of an RSS-like application over Matrix could join a room for each blog, and their reader would receive updates as new posts are made and published to any of those Matrix rooms.

I haven't yet thought about what this would actually look like in the spec in much detail, but what do you guys think?

**Comment by t3chguy** [comment]
how would it work in regards to federation and access control?

**Comment by jfrederickson** [comment]
Yeah, that's tricky - for a server to know who's allowed to change room state, it has to know which users have which power levels... hmm.

In the case of a room with hidden participants, would it make sense to have an "unknown users" option in the room permissions? So for example:

"Unknown users have power level: <dropdown, default=-1>"

-1 seems like a sane default here for spam reasons; you can't proactively ban people if you don't know where they're coming from. But essentially, most users in the room would be hidden, but you'd be able to see users with higher power levels so homeservers can handle permissions properly.

**Comment by NotAFile** [comment]
Is this not implemented yet with the "guests" feature? I guess that's pull though, not push.

**Comment by jfrederickson** [comment]
Guests are treated as normal users - while it's possible to join a room as a guest, all those users are still visible in the room. This suggestion is sort of the opposite, i.e. you could join the room with your normal user account, just most users in the room wouldn't be visible.

**Comment by t3chguy** [comment]
So peeking
But it doesn't work over federation
And then there's no access control

**Comment by NotAFile** [comment]
Well, I think it's fine to not have access control for many of the use cases. The question is rather how the hs can push updates to other hs if it doesn't know who is subscribing.

**Comment by non-Jedi** [comment]
#566 is related I think.

**Comment by turt2live** [comment]
also related (if implemented at the spec level): https://github.com/vector-im/riot-web/issues/6417
